### ISSUE TITLE
Ratings not showing up

### ISSUE THREAD

**Issue by xyzfre** [jurialmunkey/plugin.video.themoviedb.helper#375]
I’m not sure when they got affected because I was on version 4.0.55 and all the ratings were working using AuraMod Skin but when I updated to 4.0.60 A lot of the ratings don’t show up

**Comment by jurialmunkey** [comment]
Ratings are all working fine for me on latest version. You might've hit your OMDb request limit - they have a hard limit of 1000 requests per day.

**Comment by xyzfre** [comment]
![F77816E2-9329-4196-B6FD-A5B8BCB85BAA](https://user-images.githubusercontent.com/49699869/99879712-89d26a80-2bdc-11eb-9d7f-5fb4a57d62d9.jpeg)
Before

**Comment by xyzfre** [comment]
![4704B030-BE21-4233-A8F3-956E360B8C29](https://user-images.githubusercontent.com/49699869/99879722-96ef5980-2bdc-11eb-8ee2-e92ac64083c5.jpeg)
After

**Comment by xyzfre** [comment]
I reverted back to 4.0.55 and they all come back

**Comment by xyzfre** [comment]
Also noticed for some TV shows the network logo doesn’t show up 
![0A16B818-DA38-4FD4-A214-D7792C6AB149](https://user-images.githubusercontent.com/49699869/99880304-acff1900-2be0-11eb-9864-d54951639fda.jpeg)
![19F208AB-183B-4C81-8B18-00C313E2D401](https://user-images.githubusercontent.com/49699869/99880306-ad97af80-2be0-11eb-9b9a-db7266e826c9.jpeg)

**Comment by xyzfre** [comment]
In the pics it’s version .55 as you can see the ratings do work sorry to bother just wanted to report a few things I noticed with the latest version really appreciate all the hard work that goes into TMDB

**Comment by xyzfre** [comment]
My apologies for all these post finally got a chance to sit down feels like I’ve been living at my job LOL,,, but also just noticed when in seasons it does not display the episode number
![02902125-35B6-40B7-8943-CCF8636C7E30](https://user-images.githubusercontent.com/49699869/99882191-a5de0800-2bec-11eb-8531-24724f9a7ee3.jpeg)
![B7DCFE8D-E9C2-4BE6-955F-EDA42533F9B7](https://user-images.githubusercontent.com/49699869/99882192-a6769e80-2bec-11eb-98fc-54e0c3122113.jpeg)

**Comment by xyzfre** [comment]
Using Kodi 18.9 on Nvidia Shield

**Comment by jurialmunkey** [comment]
I don't know what to say other than it is working perfectly for me on latest version from master. I even tried with deleting my simplecache.db to make absolutely certain I wasn't running on cached versions.

In Arctic Horizon for those exact shows:
The Mandalorian in Trakt Trending has all available ratings, info and art:
![Honeyview_screenshot288](https://user-images.githubusercontent.com/4618719/99887626-db8cfa80-2c99-11eb-9ed3-1e2eda07fc7d.jpg)

Discovery S03E01 has ratings and season/episode numbers:
![Honeyview_screenshot289](https://user-images.githubusercontent.com/4618719/99887637-f8293280-2c99-11eb-885a-2a802f21c5fe.jpg)

The missing studio logo is most likely going to be because I put _all_ the studios available in the studio field, not just the first studio for the show. Some skins use the studio name field to match against a filename, so it needs to be an exact match for filename (which a list of names won't be). For this reason, I also output the studio names split as separate properties but it requires the skin to support that approach. 

The real issue on this point, however, is that Kodi needs a much better way in the skinning engine to be able to retrieve the first item from an infolabel list -- it's the same issue with genres or writers: you wan't the full list because otherwise you miss out on information but having a full list means you can't match the name because the skinning engine doesn't give any way to split it into parts.

**Comment by xyzfre** [comment]
I did try today with AH And the ratings work properly but not with AuraMod it seems like the TVdb and Trakt ratings don’t show up ,,I figured out with the season episode numbers if I change view type they show up

**Comment by jurialmunkey** [comment]
Tmdbhelper doesn't supply tvdb ratings. Those would be coming from Skinhelper.

**Comment by jurialmunkey** [comment]
I'll have a look at the auramod code and see what it is expecting in terms of infolabels

**Comment by jurialmunkey** [comment]
Season and Episode number are not showing because the skin shows Tagline if available. If you look at your screenshots you will see it is showing the tagline instead.

Also, since ratings are working correctly in Arctic Horizon, I'm chalking that up to a skin issue too.

**Comment by xyzfre** [comment]
> Tmdbhelper doesn't supply tvdb ratings. Those would be coming from Skinhelper.

The only thing is why when reverting back to 4.0.55 brings back the missing ratings for AuraMod

**Comment by jurialmunkey** [comment]
> The only thing is why when reverting back to 4.0.55 brings back the missing ratings for AuraMod

It's definitely a skin issue because if it works in Arctic Horizon then it works everywhere. The service monitor either fills info correctly or it doesn't -- which skin is being used doesn't change that. If it was broken, it would be broken in all skins.

Likely what is happening is there is some visibility condition in the skin that is doing some comparison against ListItem.TVShowTitle and hiding/showing the ratings based upon that condition. Previously that label was not being filled when it should have been. That was fixed after v55 and now TVShowTitle is filled correctly.

**Comment by jurialmunkey** [comment]
I just installed AuraMod and ratings are working perfectly fine for me with latest TMDbHelper.

Do you actually even have the TMDbHelper service monitor turned on the AuraMod skin settings? (I can't remember what category it was under but there it is in there).

![Honeyview_screenshot294](https://user-images.githubusercontent.com/4618719/100039758-0756de80-2e5a-11eb-938f-cd6847daf5ed.jpg)

![Honeyview_screenshot292](https://user-images.githubusercontent.com/4618719/100039761-0920a200-2e5a-11eb-8015-845a62f4b0f8.jpg)

**Comment by xyzfre** [comment]
Weird ,,I really appreciate you taking your time for checking this issue with AuraMod,, maybe it’s something on my end going to try a fresh install on a different device,,TMDB Helper is a AMAZING addon

**Comment by xyzfre** [comment]
After today’s update which I should’ve tried from the beginning,, I deleted the simplecache.db and all the ratings started showing up property ,,,once again thank you for your responses it was driving me crazy lol because I like to be on the latest versions especially when it comes to TMDB Helper Thank you for all your hard work my friend

**Comment by Bulletbling** [comment]
I'm having the same issue for some ratings, except that it's for trakt ratings mostly, and they only show up when I click on the movie and go to it's information screen and wait 5-15s. Once I click back on my remote and go back to the widgets screen, the Trakt rating is briefly shown as it should, next to the tmdb and imdb ratings, and then it disappears. I've tried deleting simplecache and even downgrading tmdbhelper to the official kodi repo version 2.4 (is even worse because then this issue applies to the imdb ratings as well only showing on the movie information screen). I'm running Auramod as well with tmdbhelper latest version 4.2.10 I believe it is.

**Comment by xyzfre** [comment]
- My apologies for jumping in but have you noticed if it's happening within ￼TMDB Helper widgets or Within Seren widgets or both only asking because on my end when I cleared the simple￼￼￼.cache it fix the issue ,but I noticed with Seren widgets only I do see the Trakt ratings missing   On some Shows or Movies I think it's a Seren issue because using Fen or Venom and TMDB H on my end I get all the proper ratings ￼￼but not with Seren also using AuraMod

**Comment by Bulletbling** [comment]
> * My apologies for jumping in but have you noticed if it's happening within ￼TMDB Helper widgets or Within Seren widgets or both only asking because on my end when I cleared the simple￼￼￼.cache it fix the issue ,but I noticed with Seren widgets only I do see the Trakt ratings missing   On some Shows or Movies I think it's a Seren issue because using Fen or Venom and TMDB H on my end I get all the proper ratings ￼￼but not with Seren also using AuraMod

Interesting. I will check this out now. And what cache did you clear? If it is indeed linked to Seren only, then maybe the devs can get to the source of the problem. Like I said, I can get the ratings while using the seren widgets, but only after I click on one of the movies to get to the information page that lets you play it. Standby for testing.

**Comment by jurialmunkey** [comment]
TMDBH no longer uses simple cache, so deleting simple cache db does nothing. If deleting simple cache is fixing an issue then it is something completely unrelated to tmdbhelper and most likely means the skin you are using is relying on skin helper for ratings not TMDBH. 

TMDBH now has its own cache files in the themoviedb helper addon_data folder. However, there really shouldn't be any need to delete cache files anymore anyway. 

Trakt ratings work perfectly fine for me using Arctic Horizon with Tmdbhelper across a variety of addons and also in the library. 

I'm pretty sure this is either an issue with your skin settings, the skin itself, or potentially skin helper.

**Comment by Bulletbling** [comment]
> TMDBH no longer uses simple cache, so deleting simple cache db does nothing. If deleting simple cache is fixing an issue then it is something completely unrelated to tmdbhelper and most likely means the skin you are using is relying on skin helper for ratings not TMDBH.
> 
> TMDBH now has its own cache files in the themoviedb helper addon_data folder. However, there really shouldn't be any need to delete cache files anymore anyway.
> 
> Trakt ratings work perfectly fine for me using Arctic Horizon with Tmdbhelper across a variety of addons and also in the library.
> 
> I'm pretty sure this is either an issue with your skin settings, the skin itself, or potentially skin helper.

Since you confirmed trakt and other ratings were working with Auramod in the past, I wonder what is causing this issue now. I can get trakt ratings, but only on the more info page; they don't work on the widgets pages. If tmdbhelper is providing the Trakt rating for this skin, then it is indeed working, but only on that movie/show info page. And when I am on that info page and click on "cast & more" and go to similar movies or recommended movies, etc., the ratings that show up for all of those movies are duplicated from whatever movie I initially selected. Really weird stuff. Have any idea of what I can clear without redoing everything (weeks of work)? I've already tried downgrading and upgrading again all add-ons, uninstalling and reinstalling tmdbhelper, getting a new omdb key, clearing cache for kodi, updating the Auramod skin to a fork that has some users "improvements" (didn't notice any effect with ratings).

**Comment by jurialmunkey** [comment]
It is going to be either a skin setting issue or an issue with the skin itself. If it works in Arctic Horizon then it is working. 

Make sure you have tmdbhelper service monitor enabled in auramod skin settings. Check all the skin settings - that skin has a lot of them.

**Comment by xyzfre** [comment]
It's a weird one cause I'm using AuraMod and Trakt ratings work fine also IMDB etc..

**Comment by xyzfre** [comment]
Here’s an example



https://user-images.githubusercontent.com/49699869/104864449-0183f380-5907-11eb-91ad-9e63d0eb5b84.MOV

**Comment by xyzfre** [comment]
https://user-images.githubusercontent.com/49699869/104864943-8e7b7c80-5908-11eb-8132-fb66c8bf305c.MOV
My apologies for the poor video quality LOL

**Comment by Bulletbling** [comment]
> https://user-images.githubusercontent.com/49699869/104864943-8e7b7c80-5908-11eb-8132-fb66c8bf305c.MOV
> My apologies for the poor video quality LOL


This is what mine does. I have imdb, tmdb, and trakt set on (rotten tomatoes is biased for critics...not most people so I don't use them).

**Comment by Bulletbling** [comment]
> It is going to be either a skin setting issue or an issue with the skin itself. If it works in Arctic Horizon then it is working - no amount of reinstalling addons is going to change anything with it not working only in one specific skin.
> 
> Make sure you have tmdbhelper service monitor enabled in auramod skin settings. Check all the skin settings - that skin has a lot of them.

Looks like I can get it to work when not using any of the "Netflix" style home screens. Using the Aura home screen allows the Trakt ratings to show. I may just keep it set as aura home screen. I downloaded this auramod to have a kodi setup look clean and polished like netflix, but I can settle for the Aura home.

Btw, thanks jurialmunkey for letting us discuss this here. There's no other place to really discuss the Auramod integration with tmdbhelper or whatever else is pulling trakt ratings for Auramod since the github for Auramod has no bug reporting and the forum can't really help since I guess this seems to be a home styling issue for a mod that hasn't been updated in 6+ months.

**Comment by Bulletbling** [comment]
> Here’s an example
> 
>  63262788183__C0B3E337-C98C-434B-A676-AF53688FAA0C.MOV

I can confirm. It is working for you because you aren't using the netflix home layout. For the netflix layout, only the ratings for the spotlight view work and it'll get that spotlight movie trakt rating stuck on all other content you view within the widgets. Switching to another layout resolves the issue in Auramod so it's not a TMDB Helper issue.
